daffodil-test/src/test/resources/org/apache/daffodil/section00/general/testExpressionPropertyWarnings.tdml

```diff
@@ -0,0 +1,46 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/"
+  xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:ex="http://example.com"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions"
+  xmlns:daf="urn:ogf:dfdl:2013:imp:daffodil.apache.org:2018:ext"
+  suiteName="test of warnings for when a property can't be an expression but looks like an expression"
+  defaultRoundTrip="onePass">
+
+  <tdml:defineSchema name="S1" elementFormDefault="unqualified">
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited"/>
+    <xs:element name="root" type="xs:double" dfdl:textNumberPattern="#####0.0#####" dfdl:textStandardInfinityRep="{../ex:thisCantBeAnExpression}" />
+  </tdml:defineSchema>
+
+  <tdml:parserTestCase name="expressionPropertyWarning1" model="S1" root="root">
+    <tdml:document>1.0</tdml:document>
+    <tdml:infoset>
+      <tdml:dfdlInfoset>
+        <ex:root xsi:type="xs:double">1.0</ex:root>
```

Comment:
Why do you have an `xsi:type="xs:double"` when the schema already declares the root element with the xs:double type?

Reply:
Without that decoration, the 1.0 was reduced to 1, and the round-trip parse-unparse ended up having different results on either end.

Reply:
Hmm, I don't think Daffodil takes into account ``xsi:type`` except for a comparing expected vs actual results. And even then, it only uses it it for ``xs:anyURI`` and calendar types (e.g. ``xs:dateTime``). I'm pretty sure ``xsi:type="xs:double"`` shouldn't have an affect.

Reply:
I thought the comparison did type specific equality using xsi:type information?

Reply:
It does, but not for numeric types--only for xs:anyURI, xs:hexBinary, xs:time, and xs:dateTime. These are the only ones where IBM DFDL and Daffodil have different output so we needed a little extra smarts. We could add it for the remainder of types, but haven't had the need so far. 

https://github.com/apache/incubator-daffodil/blob/master/daffodil-lib/src/main/scala/org/apache/daffodil/xml/XMLUtils.scala#L1088-L1108

Reply:
Ah, so for now, the only way to get a test like this to work is to make sure textNumberPattern constraints the number layout in the data, and use true values in the infoset, or just use strings as the element types. 

Honestly, I think we do need this for double and float, as they are so snarky about comparison. Without type/schema, it's all strings, so, 1 and 1.0 in XML are not equivalent.

Reply:
Created DAFFODIL-2402